modulemd/include/modulemd-2.0/modulemd-module-stream.h

```diff
@@ -468,4 +469,20 @@ gboolean
 modulemd_module_stream_build_depends_on_stream (ModulemdModuleStream *self,
                                                 const gchar *module_name,
                                                 const gchar *stream_name);
+
+/**
+ * modulemd_module_stream_get_obsoletes_resolved:
+ * @self: (in): This #ModulemdModuleStream object.
+ *
+ * Returns: (transfer none): The #ModulemdObsoletes information associated with
+ * this object. If the associated obsoletes has reset attribute set, this
+ * function doesn't return it. From outside obsoletes with reset looks like
+ * there is no obsoletes set for this stream. Every obsoletes (even with reset) can
+ * be accessed from the streams module.
+ *
+ * Since: 2.10
+ */
+ModulemdObsoletes *
+modulemd_module_stream_get_obsoletes_resolved (ModulemdModuleStream *self);
+
```

Comment:
As an addendum to my comment above, that would also mean that we don't need to make this a virtual method.

Reply:
So do I understand correctly then there would be no `modulemd_module_stream_get_obsoletes_resolved` only `modulemd_module_stream_v2_get_obsoletes_resolved`? And I should move the `obsoletes` pointer from the `ModulemdModuleStreamPrivate` struct into `_ModulemdModuleStreamV2`?

Reply:
@sgallagher is the code ok in the current state in regards to `modulemd_module_stream_get_obsoletes_resolved`? (I have just dropped the v1 version) Can I squash the commits now?

Reply:
> So do I understand correctly then there would be no `modulemd_module_stream_get_obsoletes_resolved` only `modulemd_module_stream_v2_get_obsoletes_resolved`? And I should move the `obsoletes` pointer from the `ModulemdModuleStreamPrivate` struct into `_ModulemdModuleStreamV2`?

Sorry, I missed the notification for this question. Yes, this is what I meant. The `ModuleStream` object is meant to only provide the set of functions that *all* streams must have. Basically, this just means the set that are required to add them to a `ModuleIndex`. So please remove this from ModuleStream and implement it in ModuleStreamV2.

Reply:
Ok I have squashed the commits since it was getting more confusing I think, but most importantly I moved all the obsoletes from `ModuleStream` to `ModuleStreamV2`. I had to also add the automatic stream upgrading to `modulemd_module_add_obsoletes` and `modulemd_module_add_stream` since they can both associate obsoletes with a stream (also updated their docs).

Reply:
@sgallagher can you check the update from my previous comment? Or is there something else that needs fixing? If possible I would like to get this finished. :slightly_smiling_face: